cpp/src/parquet/column_reader.cc

```diff
@@ -2225,10 +2245,9 @@ class ByteArrayDictionaryRecordReader final : public TypedRecordReader<ByteArray
     } else {
       num_decoded = this->current_decoder_->DecodeArrowNonNull(
           static_cast<int>(values_to_read), &builder_);
-
-      /// Flush values since they have been copied into the builder
-      ResetValues();
```

Comment:
This is a logical change but I don't think it would matter 😅

Reply:
Sounds reasonable to me.